fiat/prices.go

```diff
@@ -11,11 +11,53 @@ import (
 )
 
 var (
-	errNoPrices        = errors.New("no price data provided")
-	errPriceOutOfRange = errors.New("timestamp before beginning of price " +
+	errNoPrices            = errors.New("no price data provided")
+	errUnknownPriceBackend = errors.New("unknown price backend")
+	errPriceOutOfRange     = errors.New("timestamp before beginning of price " +
 		"dataset")
+
+	// errGranularityRequired is returned when a request is made that
+	// required fiat prices but the granularity of those prices is not set.
+	errGranularityRequired = errors.New("granularity required when " +
+		"fiat prices are enabled")
 )
 
+// PriceAPIBackend is an interface that must be implemented by any backend that
+// is used to fetch fiat price information.
+type PriceAPIBackend interface {
+	GetPrices(ctx context.Context, startTime, endTime time.Time) ([]*USDPrice, error)
+}
+
+type PriceBackend string
```

Comment:
nit: godocs for this enum and the coincap/coindesk values below.

I also have a _slight_ preference for uint8 for categories like this, I think it's a bit more standard, but that's a loose opinion loosely held. If you just do ints, you can add a `String` method which returns the names as you have them here.